lib/cocoapods-core/source.rb

```diff
@@ -326,6 +369,18 @@ def specs_dir
       end
     end
 
+    def git_commit_hash
+      (`git rev-parse HEAD` || '').strip
+    end
+
+    def update_git_repo
```

Comment:
You're right, this prints after command is finished. So what should we do? 

I think we're gonna remove `CoreUI.puts output if show_output` part in update method here in order not to have duplicate output. However, I didn't fully get how to override a method of Source class in cocoapods code (probably in sources_manager.rb but how)?

I'm not a native ruby person so probably I miss sth :)